### DISCUSSION TITLE
veilid shares some goals with reticulum

### DISCUSSION THREAD

**Discussion by khimaros** [markqvist/Reticulum#361] [Ideas]
launched today at DEFCON by the Cult of the Dead Cow. it seems to be trying to solve similar problems and also operates at a low level similar to where Reticulum sits. i thought it worth mentioning and i'm interested to know what you all think of the project and where it differs from Reticulum/NomadNet.

https://veilid.com/Launch-Slides-Veilid.pdf

**Comment by faragher** [comment]
Love it! Haven't dug too deeply into the nuts and bolts, but I'd say both have a broadly similar concept and encryption, 

The major difference is in the . . . look, I don't mean values in a moral sense, they're highly compatible in that regard, but their values are different in what they believe is important.

Veilid remains on TCP and is highly extensible. That's more than fine, it's probably 99% of use cases. Reticulum places less value on doing everything and is a ground-up network stack that highly values efficiency and can run effectively over packet radio.

Both have their place, and while I think Veilid may do some thing better than Reticulum, I think Reticulum can handle more use cases.

I do like their rapid search algorithms, though, as I strongly believe one of Reticulum's major avenues for improvement is memory usage, and moving the routing tables from RAM to a rapidly searchable file system (with a most-used cache in memory) would make a considerable difference. However, I am not the person to provide a specific suggestion or implement such a system.

All in all, I like seeing this kind of thing. More options is good in the short term, and I think the implementations are all young enough that more ideas and testing are more important than interoperability. Since we can all bridge to TCP, I doubt that's too terrible an issue anyway. Could you imagine different network protocols and even physical networks all talking to each other? It's a beautiful mess.

**Reply by khimaros** [reply]
thank you for the thoughtful reply! in the slide deck they mentioned that there is a UDP implementation but i was still not sure how much idle traffic and overhead veilid produces. maybe an area for further inquiry?

**Reply by joakim** [reply]
Sorry for necroposting, I just wanted to add my observations. From their [DefCon slides](https://veilid.com/Launch-Slides-Veilid.pdf):

> A single initial DNS TXT record request returns some bootstrap nodes that are known to exist. Those are asked to return nodes that are ‘close’ to your own node.

A single point of failure, sounds like to me. I wouldn't want to rely on DNS, which is to a large degree centralized and an obvious target for attacks by powerful actors.

I'm also skeptical of protocols that do too much (typically involving DHTs), narrowing the possibility space. I prefer the Unix philosophy of doing one thing well and designing for composability. Reticulum only does networking, I think that's wise.

Besides those nitpicks, I like Veilid! There's room for both Reticulum and Veilid, they do things differently and have different strengths.

> Could you imagine different network protocols and even physical networks all talking to each other? It's a beautiful mess.

Well said! I'd love to see a beautiful mess of privacy-respecting protocols filling the void :)
